lavalab-gen.py: Permit to customize exported ports to a slave

montjoie opened the pull request:
This patch adds a way to choose a number of ports to be exported from
host to slave.

montjoie committed:
lavalab-gen.py: Permit to customize exposed ports to a slave

This patch adds a way to choose a number of ports to be exposed from
host to slave.

khilman commented:
I think the YAML format should be a little different, and I think "export" is a bit confusing as a term.  I would say 
```
expose_ports:
  - p1:p2
  - p3:p4
```
Which matches the `src:dest` format used by the docker files and command line.

khilman commented on README.md:
```diff
@@ -247,6 +247,8 @@ slaves:
     remote_proto:		http(default) or https
     default_slave:		Does this slave is the default slave where to add boards (default: lab-slave-0)
     export_ser2net:		Do ser2net ports need to be available on host
+    export_port:		Export ports p1 on the host to p2 on the worker slave.
```
Hmm, as I think about it, "export" isn't the right word either, your first patch was better.  I think "expose" is better because that's the term docker docs use.   So..

```
expose_ports:  Expose container port p2 to port p1 on the host`
 - p1:p2
```

khilman commented:
Can you make this plural?  e.g. `expose_ports` instead of `expose_port`

khilman commented:
And then add another patch to change `export_ser2net` to `expose_ser2net`.   **BUT**, please keep the original in the code (but not the README) for backward compatibility.

montjoie committed:
Rename export_ser2net to expose_ser2net

This patch rename export_ser2net to expose_ser2net.
The old export_ser2net is still handled but we now print a deprecating
message.
